text/0715-construct-upgrade-support.md

```diff
@@ -0,0 +1,1251 @@
+|| ┌───────────────────┐ |         |┌─────────────────────┐||                                 
+|| │ Execute Stack     | |         |│     Execute         |||  
+|| |    refactoring    | |         || Retain-Remove-Import||| 
+|| └───────────────────┘ |         |└─────────────────────┘|| 
+|└───────────────────────┘         └───────────────────────┘|
+|           │                               │               | 
+|           └───────────────────────────────┘               |
+|                          |                                |
+|                          ▼                                |
+|               ┌─────────────────────┐                     |
+|               │        Done         │                     |
+|               └─────────────────────┘                     |
+└───────────────────────────────────────────────────────────┘
+```
+
+## Settings
+
+You have a few settings available to control the behavior of the new CDK CLI command
+for construct upgrade feature.
+
+```sh
+cdk construct-upgrade --unstable=construct-upgrade [stacks] \
+  --target [id] \
+  [OPTIONS]
+```
+
+Arguments:
+
+- CDK stack ID (Optional)
+    The construct ID of the CDK stack from your app to synthesize.
+
+Options:
+
+- --target (Required):
+  The target construct and module you want to upgrade to. For example, `aws-cdk-lib.aws-dynamodb.TableV2` or `@aws-cdk/aws-ec2.VpcV2`
+  For fully supported construct upgrades, we can short hand it to `TableV2` or `VpcV2`.
+- --dry-run (Optional)
+  This flag is default to false. When the flag is set to true, the CLI acts as dry-run and will only trigger validation.
+- --require-approval (Optional)
+  Specify what security-sensitive changes require manual approval.
+  - any-change – Manual approval required for any change to the stack.
+  - broadening – Manual approval required if changes involve a broadening of permissions or security group rules.
+  - never – Approval is not required.
+  Valid values: [any-change, broadening, never]
+  Default value: broadening
+- --skip-validation (Optional)
+  Default to false. Use as users' own risk to by pass the validations step.
+- --custom-analysis (Optional)
+  When user provides a custom JavaScript file for validation, more will be explained in [Custom Validations](#custom-validations-optional-future-enhancement)
+- Other global options like --quiet, --help, etc
+
+## Construct Upgrade Strategies
+
+The CDK construct upgrade framework employs two different strategies for migrating resources, automatically
+selecting the appropriate approach based on the `--target` construct. CDK CLI stores and uses a context variable
+to determine the mapping between upgrade targets and their corresponding strategies.
+
+### In-place Migration
+
+This strategy is suitable for constructs where changes do not involve custom resource
+removal or introduction of new underlying resource types. It leverages the CloudFormation stack refactoring
+feature to maintain resource continuity.
```

Comment:
This could be clearer on the problem it solves, "maintain resource continuity" is a bit vague. Today moving constructs around and similar changes in a CDK application can cause the logical id of the resulting cloud-formation resource to change, resulting in a deletion and recreation of the resources. Stack refactoring enables a logical name change without any change to the deployed resource.